internal/versions/versions.go

```diff
@@ -15,6 +15,11 @@ var (
 	// PluginOrderingVersionCutoff is the Kong version prior to the addition of plugin ordering.
 	PluginOrderingVersionCutoff = semver.Version{Major: 3}
 
+	// ConsumerGroupsVersionCutoff is the Kong version prior to the addition of Consumer Groups as first class citizens.
+	// TODO: Change to 3.4 once it's released. For now let's assume it's 3.3 (some features work), but for the full
+	// support it should be 3.4 (unreleased yet).
```

Comment:
IMO we should make it `3.4` straight away given that we want to officially support only the new ConsumerGroups Kong's implementation (that's also how it's announced in 2.11 blogpost).

Reply:
Yeah, but when I do it now I wouldn't be able to test that small part introduced in this PR. It should work the same way in both versions I mean with Kong 3.3 and 3.4. I put this guard to not run this test against older gateways that it may not pass. Do you have an idea for a better approach to handle such a situation? Because the solution I proposed here is not something I consider a great 😄

Reply:
Ah, I didn't catch it's only used in tests. Can we create an issue that would track changing this and extending tests?

BTW is there possibly some release candidate of Kong 3.4 that we could use to write those extended tests before we release KIC 2.11?

Reply:
Good point. I've added it to https://github.com/Kong/kubernetes-ingress-controller/issues/4411, I have to check for release candidates

Reply:
I've tried with `3.4` and
```
NAME  TestConsumerGroup
    consumer_group_test.go:27: skipping because Kong version "3.3.1.0" is not within test's range ">=3.4.0": 
--- SKIP: TestConsumerGroup (1.01s)
```
we're using `kong/kong-gateway-internal:master-alpine` image in CI for PRs. Image `kong/kong-gateway-dev:latest` presents version `3.4.0.0` in the header, maybe it should be used in CI to have bleeding-edge Gateway... 🤔

Reply:
To not block this PR I'd say let's stick to the status-quo - make the cutoff version to be 3.3 and do not modify the version we test against in CI yet.

Reply:
I mentioned about it in https://github.com/Kong/kubernetes-ingress-controller/issues/4411